cli/template/extras/src/server/api/trpc/with-auth.ts

```diff
@@ -46,10 +46,8 @@ const createInnerTRPCContext = (opts: CreateContextOptions) => {
  * @see https://trpc.io/docs/context
  */
 export const createTRPCContext = async (opts: CreateNextContextOptions) => {
```

Comment:
In my opinion, we should destructure wherever possible because its one less variable to think about, one less variable to name. So I think destructuring in the first change is fine. Here I would prefer if we destructure `opts` into `req` and `res` in the function params itself.

Reply:
In case of destructuring here, we would have to think about two variables `req` and `res` instead of one `opts`.

However, passing destructured `req` and `res` as parameters can make the code more explicit, as it clearly shows that the getServerAuthSession function requires both of them. 

Therefore, I'll go ahead and destructure `req` and `res` directly in the function's parameter list.

Reply:
Fixed.

Reply:
Fixed

Reply:
> In case of destructuring here, we would have to think about two variables `req` and `res` instead of one `opts`.
no not really, those are properties coming from the opts object. they aren't variables we declared